From 2c7ca88e7c09e36477c5b92a37716262dc837bfd Mon Sep 17 00:00:00 2001
From: Jason Liu <jason.liu@gumstix.com>
Date: Mon, 12 Jun 2017 09:59:21 -0700
Subject: [PATCH 3/5] iio: fix whoami for accel,mag,gyro

Fix the whoami problem for st_accel, st_mag, st_gyro driver

Signed-off-by: Jason Liu <jason.liu@gumstix.com>
---
 drivers/iio/accel/st_accel_core.c       | 2 +-
 drivers/iio/gyro/st_gyro.h              | 1 +
 drivers/iio/gyro/st_gyro_core.c         | 2 +-
 drivers/iio/gyro/st_gyro_i2c.c          | 9 +++++++++
 drivers/iio/gyro/st_gyro_spi.c          | 1 +
 drivers/iio/magnetometer/st_magn_core.c | 2 +-
 6 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/drivers/iio/accel/st_accel_core.c b/drivers/iio/accel/st_accel_core.c
index da3fb06..abb7813 100644
--- a/drivers/iio/accel/st_accel_core.c
+++ b/drivers/iio/accel/st_accel_core.c
@@ -44,7 +44,7 @@
 #define ST_ACCEL_FS_AVL_400G			400
 
 /* CUSTOM VALUES FOR SENSOR 1 */
-#define ST_ACCEL_1_WAI_EXP			0x33
+#define ST_ACCEL_1_WAI_EXP			0x49
 #define ST_ACCEL_1_ODR_ADDR			0x20
 #define ST_ACCEL_1_ODR_MASK			0xf0
 #define ST_ACCEL_1_ODR_AVL_1HZ_VAL		0x01
diff --git a/drivers/iio/gyro/st_gyro.h b/drivers/iio/gyro/st_gyro.h
index a5c5c4e..48923ae 100644
--- a/drivers/iio/gyro/st_gyro.h
+++ b/drivers/iio/gyro/st_gyro.h
@@ -19,6 +19,7 @@
 #define LSM330DL_GYRO_DEV_NAME		"lsm330dl_gyro"
 #define LSM330DLC_GYRO_DEV_NAME		"lsm330dlc_gyro"
 #define L3GD20_GYRO_DEV_NAME		"l3gd20"
+#define L3GD20H_GYRO_DEV_NAME		"l3gd20h"
 #define L3G4IS_GYRO_DEV_NAME		"l3g4is_ui"
 #define LSM330_GYRO_DEV_NAME		"lsm330_gyro"
 #define LSM9DS0_GYRO_DEV_NAME		"lsm9ds0_gyro"
diff --git a/drivers/iio/gyro/st_gyro_core.c b/drivers/iio/gyro/st_gyro_core.c
index aea034d..e504676 100644
--- a/drivers/iio/gyro/st_gyro_core.c
+++ b/drivers/iio/gyro/st_gyro_core.c
@@ -269,7 +269,7 @@ static const struct st_sensor_settings st_gyro_sensors_settings[] = {
 		.wai = ST_GYRO_3_WAI_EXP,
 		.wai_addr = ST_SENSORS_DEFAULT_WAI_ADDRESS,
 		.sensors_supported = {
-			[0] = L3GD20_GYRO_DEV_NAME,
+			[0] = L3GD20H_GYRO_DEV_NAME,
 		},
 		.ch = (struct iio_chan_spec *)st_gyro_16bit_channels,
 		.odr = {
diff --git a/drivers/iio/gyro/st_gyro_i2c.c b/drivers/iio/gyro/st_gyro_i2c.c
index 40056b8..d15cca3 100644
--- a/drivers/iio/gyro/st_gyro_i2c.c
+++ b/drivers/iio/gyro/st_gyro_i2c.c
@@ -41,6 +41,14 @@ static const struct of_device_id st_gyro_of_match[] = {
 		.data = L3GD20_GYRO_DEV_NAME,
 	},
 	{
+		.compatible = "st,l3gd20h-gyr",
+		.data = L3GD20H_GYRO_DEV_NAME,
+	},
+	{
+		.compatible = "st,l3gd20h-gyro",
+		.data = L3GD20H_GYRO_DEV_NAME,
+	},
+	{
 		.compatible = "st,l3g4is-gyro",
 		.data = L3G4IS_GYRO_DEV_NAME,
 	},
@@ -98,6 +106,7 @@ static const struct i2c_device_id st_gyro_id_table[] = {
 	{ L3G4IS_GYRO_DEV_NAME },
 	{ LSM330_GYRO_DEV_NAME },
 	{ LSM9DS0_GYRO_DEV_NAME },
+	{ L3GD20H_GYRO_DEV_NAME },
 	{},
 };
 MODULE_DEVICE_TABLE(i2c, st_gyro_id_table);
diff --git a/drivers/iio/gyro/st_gyro_spi.c b/drivers/iio/gyro/st_gyro_spi.c
index fbf2fae..b41bff5 100644
--- a/drivers/iio/gyro/st_gyro_spi.c
+++ b/drivers/iio/gyro/st_gyro_spi.c
@@ -55,6 +55,7 @@ static const struct spi_device_id st_gyro_id_table[] = {
 	{ L3G4IS_GYRO_DEV_NAME },
 	{ LSM330_GYRO_DEV_NAME },
 	{ LSM9DS0_GYRO_DEV_NAME },
+	{ L3GD20H_GYRO_DEV_NAME },
 	{},
 };
 MODULE_DEVICE_TABLE(spi, st_gyro_id_table);
diff --git a/drivers/iio/magnetometer/st_magn_core.c b/drivers/iio/magnetometer/st_magn_core.c
index 3e1f06b..de68c9d 100644
--- a/drivers/iio/magnetometer/st_magn_core.c
+++ b/drivers/iio/magnetometer/st_magn_core.c
@@ -87,7 +87,7 @@
 #define ST_MAGN_0_MULTIREAD_BIT			false
 
 /* CUSTOM VALUES FOR SENSOR 1 */
-#define ST_MAGN_1_WAI_EXP			0x3c
+#define ST_MAGN_1_WAI_EXP			0x49
 #define ST_MAGN_1_ODR_ADDR			0x00
 #define ST_MAGN_1_ODR_MASK			0x1c
 #define ST_MAGN_1_ODR_AVL_1HZ_VAL		0x00
-- 
2.7.4

